### DISCUSSION TITLE
How to add custom event listener in theia

### DISCUSSION THREAD

**Discussion by 0xPiranhaCodes** [eclipse-theia/theia#8973] [Q&A] [answered]
If I want to add a function like the below how can I do it in the current version of theia

`window.addEventListener('load', function() {
            window.ideCallback();
          });`

In the oldest version, I version I was adding this in the index.html file

**Comment by vince-fugnitto** [comment, marked as answer]
@satheesh1997 I'm not sure the specifics of what you are trying to achieve, but I believe the goal would be to create a custom extension with proper events and event handlers.

- https://theia-ide.org/docs/authoring_extensions/

For events and event handlers it would just be the standard TypeScript/JavaScript syntax.

**Reply by 0xPiranhaCodes** [reply]
Thank you :)
